src/lib/ldap/connection.c

```diff
@@ -365,6 +363,7 @@ static fr_connection_state_t _ldap_connection_init(void **h, fr_connection_t *co
 	 */
 	MEM(c->queries = fr_rb_inline_talloc_alloc(c, fr_ldap_query_t, node, fr_ldap_query_cmp, NULL));
 	fr_dlist_init(&c->refs, fr_ldap_query_t, entry);
+	fr_connection_add_watch_pre(conn, FR_CONNECTION_STATE_CLOSED, _ldap_connection_close_watch, true, c);
```

Comment:
I wouldn't expect this to help because I don't think the trunk code gets a chance to run before the watch is added?

Reply:
@arr2036 this patch helps.

The problem is that in case of error the close watch remains and is called on a free'd ldap_conn.